### ISSUE TITLE
Cannot open ROS2 bags, buffer stream error.

### ISSUE THREAD

**Issue by iandareid** [facontidavide/PlotJuggler#873]
Recently, I opened PlotJuggler and when I went to open a ROS2 bag I had opened before with PlotJuggler, it gave me the following:

![Screenshot from 2023-10-11 13-43-55](https://github.com/facontidavide/PlotJuggler/assets/107203645/62434656-8d3a-4266-9ff3-9adda46e3418)

I have previously opened this bag and it presented no issues. I did discover that it works fine as long as I do not load more than 4 topics. I could not recreate with the dummy data but I have uploaded the bag, This does not seem to depend on size, just 4 topics.

I have restarted my machine, uninstalled and purged my system of all PlotJuggler related files and reinstalled to no avail. I thought that maybe previous data was stuck in a cache somewhere, but I am unsure since I purged PlotJuggler and ran a grep on the entire system and nothing but apt logs and dpkg backups showed up.

I am running Ubuntu 22.04 and ROS2 Humble.

To reproduce:

1. Select load data from file button.
2. Select bag.
3. Select desired topics
4. Press enter.

**Comment by iandareid** [comment]
I cloned the repo and tried to track down where the error comes from and found that it has to do with the cdrBuffer. I don't actually know what this is but, it appears to be related.

**Comment by iandareid** [comment]
Turns out the issue was I had local changes to a message definition and not on the device that I recorded the bag on. This error message is still unclear. My current theory (based on a quick survey of the code) is that since I made the message larger, the length of each message messed with the buffer stream, so it did not present as a unrecognized message type. 

I hope this helps the next person to run into this issue.
